addons/l10n_in_check_gst_status/models/res_partner.py

```diff
@@ -0,0 +1,47 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+from odoo.addons.iap import jsonrpc
+from odoo.exceptions import UserError, AccessError
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    l10n_in_gstin_verified_status = fields.Char(string="GSTIN Verified Status", tracking=True)
+    l10n_in_gstin_verified_date = fields.Date(string="GSTIN Verified Date")
+
+    def _gstin_status_get_baseurl(self):
+        return "https://jva-odoo-iap-apps2-15-0-gst-number-active-check-jva-13181640.dev.odoo.com/iap/l10n_in_reports/1/public/search"
+
+    def get_verified_status(self):
+        url = self._gstin_status_get_baseurl()
+        user_token = self.env["iap.account"].get("l10n_in_edi")
+        uuid = self.env["ir.config_parameter"].sudo().get_param("database.uuid")
+        params = {"account_token": user_token.account_token, "dbuuid": uuid, "gstin_to_search": self.vat}
+        response = self._fetch_api_response(url, params)
+        self._assign_values(response)
+
+    def _fetch_api_response(self, url, params):
+        try:
+            response = jsonrpc(url, params=params, timeout=25)
+            return response
+        except AccessError as e:
+            raise UserError(_("Error: %s", str(e)))
+
+    def _assign_values(self, response):
+        if response.get("error"):
+            error = response["error"]
+            error_codes = [e.get("code") for e in error]
+            if any(code in ['SWEB_9035', 'FO8000'] for code in error_codes):
+                self.l10n_in_gstin_verified_status = "Invalid"
+                self.l10n_in_gstin_verified_date = fields.Date.today()
+            elif "GEN5008" in error_codes:
+                raise UserError(_("The GSTIN status verification service is currently undergoing maintenance. Please try again later."))
+            else:
+                raise UserError(_("Error: %s", str(error)))
+        else:
```

Comment:
```suggestion
```
No need for else might be better to move the error handling into different method may be